ecommerce/views.py

```diff
@@ -20,40 +21,54 @@
 from django.template.loader import render_to_string, get_template
 from django.core.paginator import Paginator
 from django.utils.translation import gettext_lazy as _
+from rest_framework import generics
+from rest_framework.views import APIView
+from django.shortcuts import render
+from rest_framework.response import Response
+from rest_framework import status
 from .models import Booking, Cart, Category, Comment, FavoriteProduct, Order, Product, Review
 
-def product_get(request):
-    products = Product.objects.all()
-    categories = Category.objects.all()
-    if request.user.is_authenticated:
-        favorite_product_ids = FavoriteProduct.objects.filter(user=request.user).values_list('product_id', flat=True)
-    else:
-        favorite_product_ids=[]
-
-    context = {
-        'category_list':categories,
-        'favorite_product_ids':favorite_product_ids,
-        "page_obj":products
-    }
-    return render(request, 'ecommerce/product_list.html',context=context)
-
-def product_detail(request, pk):
-    product = Product.objects.filter(pk=pk).first()
-    review = Review.objects.filter(product__pk=pk)
-    pd = Booking.objects.select_related('order').filter(product__pk=pk)
-    paid = True if pd else False
-    review_form = ReviewForm()
-    cart_form = CartForm(initial={'quantity': 0})
-
-    context = {
-        'product':product,
-        'review':review,
-        'paid':paid,
-        'review_form':review_form,
-        'cart_form':cart_form
-    }
-
-    return render(request,'ecommerce/product_detail.html',context=context)
+# def product_get(request):
+#     products = Product.objects.all()
+#     categories = Category.objects.all()
+#     if request.user.is_authenticated:
+#         favorite_product_ids = FavoriteProduct.objects.filter(user=request.user).values_list('product_id', flat=True)
+#     else:
+#         favorite_product_ids=[]
+
+#     context = {
+#         'category_list':categories,
+#         'favorite_product_ids':favorite_product_ids,
+#         "page_obj":products
+#     }
+#     return render(request, 'ecommerce/product_list.html',context=context)
+
+
+class ProductList(generics.ListCreateAPIView):
```

Comment:
Viết mới không đụng gì đến view cũ nha e
```suggestion
class APIProductList(generics.ListCreateAPIView):
```